#each joint name reffers to a joint motor
#motion names reffers to the motions which we can perform using these motors
#each motion has step value for each motor 

#motor_1: 109: left hand shouler lift(moving hand towards or away from body)
#motor_2: 106: left hand roll in vertical plane 
#motor_3: 108: left hand yaw about axix through the hand
#motor_4: 107: left hand elbow joint lift

#motor_5: 102: right hand shouler lift(moving hand towards or away from body)
#motor_6: 103: right hand roll about horizontal axis
#motor_7: 104: right hand yaw about axix through the hand
#motor_8: 105: right hand elbow joint lift


#neck_pitch : 4
#neck_roll: 2
#neck_yaw: 3


myjoints:
 names: [joint1,joint2,joint3,joint4,joint5,joint6,joint7,rock,hello_pose,joint10,left_hand_intro_final,left_hand_home,right_hand_intro_final,right_hand_home,head_left_intro,head_right_intro,head_home,attention,thinking,weather,thankyou,group_greeting,fold_both_elbow,move_forward,rotate_both_outwards,lift_both_outwards,move_neck,talking_hand_1_0,talking_hand_1_1,talking_hand_1_2,talking_hand_1_3,talking_hand_1_4,talking_hand_1_5,talking_hand_1_6,talking_hand_1_7]
 joint1:
    names: [motor_8,motor_4]
    motion:
      names: [l_1, l_2,l_3,l_4]
      l_1:
        step: [11,-8.05]  # raised hand side ways
        time_from_start: 0.75 # sec
      l_2:
        step: [0.9,2.07] #hand at rest 
        time_from_start: 1.5
      l_3:
        step: [11,-8.05]  # raised hand side ways
        time_from_start: 2.25  # sec
      l_4:
        step: [0.9,2.07]  # raised hand side ways
        time_from_start: 3 # sec
        
 joint2:
    names: [motor_2]
    motion:
      names: [m_1, m_2, m_3,m_4]
      m_1:
        step: [-49]  # radian this one was good 
        time_from_start: 10.0  # sec
      m_2:
        step: [0] 
        time_from_start: 20.0
      m_3:
        step: [43]
        time_from_start: 30.0
      m_4:
        step: [0]
        time_from_start: 40.0
 joint3:
    names: [motor_4]
    motion:
      names: [m_5,m_6]
      m_5:
        step: [-5.747]  # radian
        time_from_start: 4.0  # sec
      m_6:
        step: [4.669]
        time_from_start: 8.0
 joint4:
    names: [motor_3]
    motion:
      names: [m_7,m_8,m_9,m_10]
      m_7:
        step: [66]  # radian
        time_from_start: 10.0  # sec
      m_8:
        step: [6]
        time_from_start: 20.0
      m_9:
        step: [-26]
        time_from_start: 26.0
      m_10:
        step: [6]
        time_from_start: 32.0
 joint5:
    names: [motor_1,motor_3,motor_4]
    motion:
      names: [m_11,m_12,m_13,m_14,m_15]
      m_11:
        step: [-10,-26,4.669]  # radian
        time_from_start: 4.0  # sec
      m_12:
        step: [-10,-26,-5.747]
        time_from_start: 6.0
      m_13:
        step: [-10,-26,4.669]
        time_from_start: 8.0
      m_14:
        step: [-10,-26,-5.747]
        time_from_start: 10.0
      m_15:
        step: [2,6,4.669]
        time_from_start: 14.0
 joint6:
    names: [motor_5]
    motion:
      names: [m_16, m_17]
      m_16:
        step: [-10]  # raised hand side ways
        time_from_start: 4.0  # sec
      m_17:
        step: [1] #hand at rest 
        time_from_start: 7.0
 joint7:
    names: [motor_6]
    motion:
      names: [m_18, m_19, m_20,m_21]
      m_18:
        step: [-49]  # radian this one was good 
        time_from_start: 10.0  # sec
      m_19:
        step: [-2.37] 
        time_from_start: 20.0
      m_20:
        step: [43]
        time_from_start: 30.0
      m_21:
        step: [-2.37]
        time_from_start: 40.0
 rock:
    names: [motor_2,motor_4,motor_5,motor_6,motor_7,motor_8] # ,neck_pitch
    motion:
      names: [m_22,m_23,m_40,m_41]
      m_22:
        step: [-12.34,2.07,2.07,-10.81,-0.07,0.923] #  ,-0.177
        time_from_start: 1.0
      m_23:
        step: [-12.34,-7.74,2.07,-10.81,-0.07,11.04]  #  ,-0.177
        time_from_start: 2.0
      m_40:
        step: [-12.34,-7.74,-0.07,-10.81,2.99,11.04]  # radian  ,-0.68
        time_from_start: 3.0  # sec
      m_41:
        step: [-0.07,2.07,2.07,-2.37,-0.07,0.923] #  ,-0.177
        time_from_start: 4.0
 hello_pose: #103,104,105,4
    names: [motor_6,motor_7,motor_8 ,neck_pitch] # 
    motion:
      names: [m_24,m_25,m_26,m_27,m_33]
      m_24:
        step: [-10.81,3.14,6.062, -0.227]  # radian 
        time_from_start: 1.0  # sec
      m_25:
        step: [-10.81,3.14,7.596,-0.53]  #
        time_from_start: 1.5
      m_26:
        step: [-10.81,3.14,6.062,-0.227]  #
        time_from_start: 2.0
      m_27:
        step: [-10.81,3.14,7.596,-0.53]  #  
        time_from_start: 2.5
      m_33:
        step: [-2.37,-0.07,0.923,-0.177]  #
        time_from_start: 3.5
 joint10:
    names: [motor_1,motor_2,motor_3,motor_4,motor_5,motor_6,motor_7,motor_8] #  ,neck_roll,neck_yaw,neck_pitch
    motion:
      names: [m_28,m_29,m_30,m_31,m_32,m_61,m_62,m_63,m_64,m_65]
      m_28:
        step: [2.07,-0.07,-0.07,-8.05,2.07,-2.37,-0.07,11.04]  # radian  ,1.00,-1.34,-0.177
        time_from_start: 2.0  # sec
      m_29:
        step: [0,-0.07,-0.07,-8.05,0,-2.37,-0.07,11.04] #  ,1.00,-1.34,-0.177
        time_from_start: 4.0
      m_30:
        step: [0,-0.07,1.46,-8.05,0,-2.37,1.46,11.04] #  ,1.00,-1.34,-0.177
        time_from_start: 6.0
      m_31:
        step: [0,-6.97,1.46,-8.05,0,-9.27,1.46,11.04]  # ,1.00,-1.34,-0.177
        time_from_start: 8.0
      m_32:
        step: [0,-6.97,1.46,-8.05,0,-9.27,1.46,11.04]  # ,0.938,-1.684,-0.720
        time_from_start: 10.0
      m_61:
        step: [0,-6.97,1.46,-8.05,0,-9.27,1.46,11.04] # ,0.938,-1.070,-0.720
        time_from_start: 12.0
      m_62: 
        step: [0,-6.97,1.46,-8.05,0,-9.27,1.46,11.04]  #  ,1,-1.34,-0.177
        time_from_start: 14.0
      m_63:
        step: [0,0,1.46,-8.05,0,-2.37,1.46,11.04] #  ,1,-1.34,-0.177
        time_from_start: 16.0
      m_64:
        step: [2.07,0,1.46,-8.05,2.07,-2.37,1.46,11.04] #  ,1,-1.34,-0.177
        time_from_start: 18.0
      m_65:
        step: [2.07,0,1.46,2.07,2.07,-2.37,1.46,0.923] #  ,1,-1.34,-0.177
        time_from_start: 20.0  
 left_hand_intro_final: 
    names: [motor_2,motor_4,motor_3,motor_1] #106,107,108,109
    motion:
      names: [m_102,m_103]
      m_102:
        step: [-10.81,-8.05,0,2.07]  
        time_from_start: 1.0  
      m_103:
        step: [-10.81,-8.05,4.528,-3.14]  
        time_from_start: 2.0
 left_hand_home: 
    names: [motor_2,motor_4,motor_3,motor_1] #106,107,108,109
    motion:
      names: [m_104]
      m_104:
        step: [0,0.7,0,2.07]  
        time_from_start: 1.0 
 right_hand_intro_final: 
    names: [motor_5,motor_6,motor_7,motor_8] #102,103,104,105
    motion:
      names: [m_105,m_106]
      m_105:
        step: [2.07,-13.11,0,11.04]  
        time_from_start: 1.0  
      m_106:
        step: [-3.14,-13.11,4.528,11.04]  
        time_from_start: 2.0
 right_hand_home: 
    names: [motor_5,motor_6,motor_7,motor_8] #102,103,104,105
    motion:
      names: [m_107]
      m_107:
        step: [2.07,-2.37,0,0.923]  
        time_from_start: 1.0 
 head_left_intro: 
    names: [neck_roll,neck_yaw,neck_pitch] #  ,neck_pitch
    motion:
      names: [m_108]
      m_108:
        step: [1.0,-1.04,-0.61]  # radian  -0.227,
        time_from_start: 0.5  # sec
 head_right_intro: 
    names: [neck_roll,neck_yaw,neck_pitch] #  ,neck_pitch
    motion:
      names: [m_109]
      m_109:
        step: [1.0,-1.64,-0.61]  # radian  -0.227,
        time_from_start: 0.5  # sec
 head_home: 
    names: [neck_roll,neck_yaw,neck_pitch] #  ,neck_pitch
    motion:
      names: [m_110]
      m_110:
        step: [1.0,-1.34,-0.394]  # radian  -0.227,
        time_from_start: 0.5  # sec
 attention: 
    names: [motor_7,motor_8,motor_4,motor_3] #102,103,104,105
    motion:
      names: [m_111]
      m_111:
        step: [6.06,11.04,-7.74,9.13]  
        time_from_start: 2  
 thinking: 
    names: [neck_yaw,neck_pitch] #  ,neck_pitch
    motion:
      names: [m_112,m_113,m_114,m_115,m_116]
      m_112:
        step: [-1.76,-0.76]  # radian  -0.227,
        time_from_start: 0.5 # sec
      m_113:
        step: [-0.84,-0.76]  # radian  -0.227,
        time_from_start: 1.5 # sec
      m_114:
        step: [-1.76,-0.76]  # radian  -0.227,
        time_from_start: 2.5 # sec
      m_115:
        step: [-0.84,-0.76]  # radian  -0.227,
        time_from_start: 3.5 # sec
      m_116:
        step: [-1.34,-0.394]  # radian  -0.227,
        time_from_start: 4 # sec
 weather: 
    names: [neck_yaw,neck_pitch] #  ,neck_pitch
    motion:
      names: [m_117,m_118]
      m_117:
        step: [-1.76,0.46]  # radian  -0.227,
        time_from_start: 0.5  # sec   
      m_118:
        step: [-1.34,-0.394]  # radian  -0.227,
        time_from_start: 1  # sec 
 thankyou: 
    names: [motor_6,motor_7,motor_8,neck_pitch] #  ,neck_pitch
    motion:
      names: [m_119,m_120,m_121]
      m_119:
        step: [-9.27,15.23,11.04,-0.394]  # radian  -0.227,
        time_from_start: 2  # sec   
      m_120:
        step: [-9.27,15.23,11.04,-0.8]  # radian  -0.227,
        time_from_start: 2.5 # sec 
      m_121:
        step: [-2.37,0,0.923,-0.394]  # radian  -0.227,
        time_from_start: 4  # sec 
 group_greeting: 
    names: [motor_7,motor_8,motor_4,motor_3] #102,103,104,105
    motion:
      names: [m_122,m_123]
      m_122:
        step: [-10.81,7.59,-4.67,-10.81]  
        time_from_start: 1
      m_123:
        step: [0,0.923,2.07,0]  
        time_from_start: 2 
 fold_both_elbow: 
    names: [motor_8,motor_4] 
    motion:
      names: [m_124]
      m_124:
        step: [11.04,-7.74]  
        time_from_start: 1
 move_forward: 
    names: [motor_6,motor_2] 
    motion:
      names: [m_125]
      m_125:
        step: [-10.04,-9.27]  
        time_from_start: 1
 rotate_both_outwards: 
    names: [motor_7,motor_3] 
    motion:
      names: [m_126]
      m_126:
        step: [-10.81,-10.81]  
        time_from_start: 1
 lift_both_outwards: 
    names: [motor_5,motor_1] 
    motion:
      names: [m_127]
      m_127:
        step: [0,0]  
        time_from_start: 1
 move_neck: 
    names: [neck_roll,neck_yaw,neck_pitch] 
    motion:
      names: [m_128,m_129,m_130,m_131,m_132,m_133,m_134,m_135,m_136]
      m_128:
        step: [1.46,-1.34,-0.394]  
        time_from_start: 0.4
      m_129:
        step: [0.69,-1.34,-0.394]  
        time_from_start: 0.8
      m_130:
        step: [1.0,-1.34,-0.394]  
        time_from_start: 1.2
      m_131:
        step: [1.0,-0.99,-0.394]  
        time_from_start: 1.6
      m_132:
        step: [1.0,-1.68,-0.394]  
        time_from_start: 2.0
      m_133:
        step: [1.0,-1.34,-0.394]  
        time_from_start: 2.4
      m_134:
        step: [1.0,-0.99,0]  
        time_from_start: 2.8
      m_135:
        step: [1.0,-0.99,-0.53]  
        time_from_start: 3.2
      m_136:
        step: [1.0,-1.34,-0.394]  
        time_from_start: 3.6
 talking_hand_1_0: 
    names: [motor_5,motor_6,motor_7,motor_8] 
    motion:
      names: [m_137]
      m_137:
        step: [0,-6.20,9.13,8.36]  
        time_from_start: 0.5
 talking_hand_1_1: 
    names: [motor_2,motor_4,motor_3,motor_1] 
    motion:
      names: [m_138]
      m_138:
        step: [-3.90,-8.05,6.06,0]  
        time_from_start: 0.5
 talking_hand_1_2: 
    names: [motor_5,motor_6,motor_7,motor_8] 
    motion:
      names: [m_139]
      m_139:
        step: [0,-6.20,15.26,9.66]  
        time_from_start: 0.5
 talking_hand_1_3: 
    names: [motor_2,motor_4,motor_3,motor_1] 
    motion:
      names: [m_140]
      m_140:
        step: [-3.90,-6.74,0,0]  
        time_from_start: 0.5
 talking_hand_1_4: 
    names: [motor_5,motor_6,motor_7,motor_8] 
    motion:
      names: [m_141]
      m_141:
        step: [0,-6.20,9.13,11.04]  
        time_from_start: 0.5
 talking_hand_1_5: 
    names: [motor_2,motor_4,motor_3,motor_1] 
    motion:
      names: [m_142]
      m_142:
        step: [-3.90,-5.28,6.06,0]  
        time_from_start: 0.5
 talking_hand_1_6: 
    names: [motor_5,motor_6,motor_7,motor_8] 
    motion:
      names: [m_143]
      m_143:
        step: [0,-6.20,3,9.66]  
        time_from_start: 0.5
 talking_hand_1_7: 
    names: [motor_2,motor_4,motor_3,motor_1] 
    motion:
      names: [m_144]
      m_144:
        step: [-3.90,-6.74,12.20,0]  
        time_from_start: 0.5
